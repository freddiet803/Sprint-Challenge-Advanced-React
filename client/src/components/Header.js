import React from 'react'

import {useDarkMode} from '../Hooks/useDarkMode.js'


const Header = () => {

    const [darkMode, setDarkMode] = useDarkMode(false);

    const toggleDark = e => {
        e.preventDefault();
        setDarkMode(!darkMode);
    }


    return (
        <div className='header'>
    <button onClick={toggleDark}>{darkMode ? 'Turn Off' : 'Turn On'}</button>

        </div>
    )
}

export default Header;