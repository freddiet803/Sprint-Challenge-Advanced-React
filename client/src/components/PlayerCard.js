 import React from 'react'


 const PlayerCard = props =>{


    return (
        <div >
        <h3>Player: {props.aPlayer.name}</h3>
        <p>Country: {props.aPlayer.country}</p>
        <p>Searches: {props.aPlayer.searches}</p>
        </div>
    )
 }
export default PlayerCard;