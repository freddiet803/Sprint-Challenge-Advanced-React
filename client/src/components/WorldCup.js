import React from 'react';
import axios from 'axios';
import Header from './Header.js';
import PlayerCard from './PlayerCard.js';


 class WorldCup extends React.Component {
    
    constructor(){
        super();
        this.state = {
            soccerPlayers: []
        }
    }

    componentDidMount(){
        
        axios.get('http://localhost:5000/api/players')
        .then(res =>{
            //console.log(res.data);
            this.setState({
                soccerPlayers: res.data
            })
            //console.log('=======================')
            //console.log(this.state.soccerPlayers)
        })
        .catch(err=>{
            console.log(err);
        })
    }

    // componentDidUpdate(){
    //     //dont seem to have a reason to use it yet as we arent manipulating state
    // }
    
    render(){
        return(
            <div>
                <Header />
                <h1>Womens World Cup Players</h1>
                {this.state.soccerPlayers.map(player => {
                    return <PlayerCard key={player.id} aPlayer={player}/>
                })}
                
            </div>
        )
    }

}

export default WorldCup;