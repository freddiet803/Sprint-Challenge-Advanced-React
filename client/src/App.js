import React from 'react';
import logo from './logo.svg';
import './App.css';
import WorldCup from './components/WorldCup.js';
function App() {


  return (
    <div className="App">
       <WorldCup />
    </div>
  );
}

export default App;
