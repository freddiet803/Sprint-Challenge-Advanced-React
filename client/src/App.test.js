import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {render} from '@testing-library/react'
import PlayerCard from './components/PlayerCard'
import Header from './components/Header'
import WorldCup from './components/WorldCup'
import {shallow, configure} from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

configure({adapter: new Adapter()});

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

test('render without crashing', () =>{
  const container = render(<App/>);
  console.log(container);
  expect(container.firstChild).toMatchSnapshot();
});

test('render without Header crashing', ()=>{
  const header = render(<Header/>);
  console.log(header);
  expect(header.firstChild).toMatchSnapshot();
});

test('render without WorldCup crashing', ()=>{
  const world = render(<WorldCup/>);
  //console.log(world);
  expect(world.firstChild).toMatchSnapshot();
});



test('render player card without crashing', () =>{
  

  const wrapper = shallow(
    <PlayerCard aPlayer={{name: 'name', country: 'U S', searches: '400', id: 2020}}/>
  )
  console.log(wrapper.debug());
  
    expect(toJson(wrapper)).toMatchSnapshot()
});
