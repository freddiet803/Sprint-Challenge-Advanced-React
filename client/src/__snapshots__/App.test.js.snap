// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render player card without crashing 1`] = `
<div>
  <h3>
    Player: 
    name
  </h3>
  <p>
    Country: 
    U S
  </p>
  <p>
    Searches: 
    400
  </p>
</div>
`;

exports[`render without Header crashing 1`] = `undefined`;

exports[`render without WorldCup crashing 1`] = `undefined`;

exports[`render without crashing 1`] = `undefined`;
